### PR TITLE
UI: Update documentation links & fix `Doc link renders` test

### PR DESCRIPTION
**Pull request by bozzo**
Some links in `README` & code comments was referring to hashicorp documentation.

Small part of #74

Add missing changes in #669

### CODE REVIEW

File: changelog/683.txt [hunk 1]
```diff
@@ -0,0 +1,3 @@
+```release-note:improvement
+ui: update documentation links in README and code comments
```
Comment (cipherboy) [inline changelog/683.txt:2]: For strictly internal documentation changes, we don't need a readme.

File: ui/tests/integration/components/form-field-label-test.js [hunk 2]
```diff
@@ -44,8 +44,6 @@ module('Integration | Component | form-field-label', function (hooks) {
     assert.dom('a').doesNotExist('docLink hidden when not provided');
     this.set('docLink', '/doc/path');
     assert.dom('.sub-text').includesText('See our documentation for help', 'Doc link text renders');
-    assert
-      .dom('a')
-      .hasAttribute('href', 'https://developer.hashicorp.com' + this.docLink, 'Doc link renders');
+    assert.dom('a').hasAttribute('href', 'https://openbao.org' + this.docLink, 'Doc link renders');
```
Comment (cipherboy) [inline ui/tests/integration/components/form-field-label-test.js:47]: Do you think it is worthwhile re-enabling the UI tests now?

Comment (bozzo) [reply ui/tests/integration/components/form-field-label-test.js:47]: I don't think so, some of them still fail, or maybe in a non-blocking way

File: changelog/683.txt (hunk 1 above)
Comment (bozzo) [reply changelog/683.txt]: Ok, i removed the commit.

**Review by JanMa — APPROVED** [review]
LGTM. Thanks again @bozzo

**Pull request merged**